### ISSUE TITLE
How to know the status of the task during volcanlo scheduling?

### ISSUE THREAD

**Issue by CharlesQQ** [volcano-sh/volcano#3029]
#### What would you like to be added:
Add status of task in job.status , so user can know  the pod scheduling status.

#### Why is this needed:
User want know whether pod in what state, like `Allocated`、`Pipelined` and so on.

**Comment by lowang-bh** [comment]
There is some difficulty because the state is reset at each scheduling cycle.

**Comment by daniel-hutao** [comment]
![image](https://github.com/volcano-sh/volcano/assets/18692628/cf0fd038-898b-4b35-bb0f-90c14a3b344b)

**Comment by lowang-bh** [comment]
relative issue: [Display all jobs and queuing order in the queue #2876](https://github.com/volcano-sh/volcano/issues/2876)

**Comment by lowang-bh** [comment]
@daniel-hutao `job.Status.TaskStatusCount` is already there and it is about the final status of pod(task).

**Comment by CharlesQQ** [comment]
> job.Status.TaskStatusCount is already there and it is about the final status of pod(task)

indeed, job.Status.TaskStatusCount is the final status of pod, I think scheduling status should be add to pod's label or Annotation, Just motify function `UpdateTaskStatus` can achieve that
https://github.com/volcano-sh/volcano/blob/04a3b1b6b6b3550e292c7d3132fc1fbf3f6932cd/pkg/scheduler/api/job_info.go#L513-L529

**Comment by lowang-bh** [comment]
> I think scheduling status should be add to pod's label or Annotation, Just motify function `UpdateTaskStatus` can achieve that,

This will cause low performance of scheduler.  Because in each scheduling cycle the task's pipelined/binded node is changed randomly and frequently calling apiserver brings low performance and high latancey.
